Automattic/wp-calypso
# Checkout: Replace "Total savings" with "Coupon" and "Credits" (4)

#### Changes proposed in this Pull Request

This modifies the review step in new checkout so that rather than the "Total savings" line item (which is confusing since it represents so many different things), we re-add a "Coupon" line item, which now has a correct amount thanks to D45080-code, and add a "Credits" line item for any credits the account has.

![new-coupon-item](https://user-images.githubusercontent.com/2036909/85058826-a7546180-b170-11ea-925c-55e9530164b1.png)

This is one step toward the changes described in https://github.com/Automattic/wp-calypso/issues/43353 although there are more tasks to be done.

Fixes https://github.com/Automattic/wp-calypso/issues/43168 
Fixes https://github.com/Automattic/wp-calypso/issues/43353

#### Testing instructions

1. Use an account that already has a plan.
1. Add an upgraded plan to your cart and visit checkout. This will create a non-coupon discount for the cart item (a proration).
1. Verify that the coupon is not displayed.
1. Verify that the credits are not displayed.
1. Add a few credits to your account and reload.
1. Verify that the credits are displayed.
1. Add a coupon to your cart.
1. Verify that the coupon is displayed and shows only the discount caused by the coupon, not the proration, and not the credits.
